src/code.c

```diff
@@ -275,7 +275,7 @@ Stat NewStatOrExpr (
     while (bodySize < CS(OffsBody))
         bodySize *= 2;
     ResizeBag(body, bodySize);
-    STATE(PtrBody) = PTR_BAG(body);
+    GAP_ASSERT(STATE(PtrBody) == PTR_BAG(body));
```

Comment:
OK, so here I was wrong: This does *not* hold...

Reply:
It took me some time to understand this; but of course `ResizeBag` can change the bag pointer *even if no GC takes place*, simply because there is not enough space left at the current location of the object. Stupid me *facepalm*

I moved the assertion now to before the ResizeBag, and restored the assignment here.